### PR TITLE
feat: use PyperClipboard if available for ptk

### LINKED ISSUE #3989: PTK clipboard integration
## xonfig

<details>

```
+------------------+----------------------+
| xonsh            | 0.9.24               |
| Git SHA          | 74543ea9             |
| Commit Date      | Oct 10 15:12:47 2020 |
| Python           | 3.8.6                |
| PLY              | 3.11                 |
| have readline    | True                 |
| prompt toolkit   | 3.0.3                |
| shell type       | prompt_toolkit       |
| pygments         | 2.7.2                |
| on posix         | True                 |
| on linux         | True                 |
| distro           | manjaro              |
| on darwin        | False                |
| on windows       | False                |
| on cygwin        | False                |
| on msys2         | False                |
| is superuser     | False                |
| default encoding | utf-8                |
| xonsh encoding   | utf-8                |
| encoding errors  | surrogateescape      |
| on jupyter       | False                |
| jupyter kernel   | None                 |
| xontrib 1        | abbrevs              |
| xontrib 2        | argcomplete          |
| xontrib 3        | autovox              |
| xontrib 4        | back2dir             |
| xontrib 5        | cmd_done             |
| xontrib 6        | hist_navigator       |
| xontrib 7        | jedi                 |
| xontrib 8        | kitty                |
| xontrib 9        | pdb                  |
| xontrib 10       | prompt_ret_code      |
| xontrib 11       | vox                  |
| xontrib 12       | voxapi               |
+------------------+----------------------+

```

</details>

## Expected Behavior

PTK is able to use [system's clipboard](https://github.com/prompt-toolkit/python-prompt-toolkit/blob/master/examples/prompts/system-clipboard-integration.py) with pyperclip library. I'd expect the same with xonsh. 

## Current Behavior

The PTK shell doesn't fillup the clipboard param.

### Traceback (if applicable)

<details>

```
traceback
```

</details>

## For community
⬇️  **Please click the 👍 reaction instead of leaving a `+1` or 👍  comment**

### PR DESCRIPTION
**Pull request by jnoortheen**
closes #3989

### CODE REVIEW

File: xonsh/ptk_shell/shell.py [hunk 1]
```diff
@@ -35,6 +35,12 @@
 from prompt_toolkit.styles import merge_styles, Style
 from prompt_toolkit.styles.pygments import pygments_token_to_classname
 
+try:
+    from prompt_toolkit.clipboard.pyperclip import PyperclipClipboard
+
+    SYS_CLIPBOARD = True
```
Comment (scopatz) [inline xonsh/ptk_shell/shell.py:41]: Maybe `HAVE_SYS_CLIPBOARD`?  Is this version gated? Like could we use something like:

```
if ptk.version >= (3, 2, 1):
  from prompt_toolkit.clipboard.pyperclip import PyperclipClipboard
  HAVE_SYS_CLIPBOARD = True
else:
  HAVE_SYS_CLIPBOARD = False

```

Comment (scopatz) [reply xonsh/ptk_shell/shell.py:41]: Or I guess it needs pyperclip?

Comment (jnoortheen) [reply xonsh/ptk_shell/shell.py:41]: The current version of ptk is 3.0.8 , xonsh requires ptk v3 already.
This needs pyperclip which is an optional dependency not installed with ptk.

Comment (scopatz) [reply xonsh/ptk_shell/shell.py:41]: right, yeah.  OK.

File: setup.py [hunk 2]
```diff
@@ -342,7 +342,7 @@ def main():
     }
     skw["cmdclass"]["develop"] = xdevelop
     skw["extras_require"] = {
-        "ptk": ["prompt-toolkit>=3.0"],
+        "ptk": ["prompt-toolkit>=3.0", "pyperclip"],
```
Comment (scopatz) [inline setup.py:345]: does it make sense to add pyperclip to full too?

Comment (jnoortheen) [reply setup.py:345]: since ptk is part of full, it will automatically get installed when full is used.

Comment (scopatz) [reply setup.py:345]: Oh right!

File: xonsh/ptk_shell/shell.py (hunk 1 above)
Comment (scopatz) [reply xonsh/ptk_shell/shell.py:41]: Also, should this be `HAVE_SYS_CLIPBOARD`?

Comment (jnoortheen) [reply xonsh/ptk_shell/shell.py:41]: Yes will do.

Comment (jnoortheen) [reply xonsh/ptk_shell/shell.py]: updated.

**Comment by scopatz** [comment]
Thanks @jnoortheen!

**Pull request merged**